distribution/lib/Standard/Database/0.0.0-dev/src/Data/Column.enso

```diff
@@ -550,26 +559,143 @@ type Column
                 op_result = self.make_op "IIF" [when_true, when_false] new_name
                 adapt_unified_column op_result common_type
 
-    ## Rounding values is not supported in database columns.
-    round : Integer -> Boolean -> Problem_Behavior -> Column | Illegal_Argument | Invalid_Value_Type
-    round self decimal_places=0 use_bankers=False on_problems=Report_Warning =
-        _ = [decimal_places, use_bankers, on_problems]
-        Error.throw <| Unsupported_Database_Operation.Error "`Column.round` is not implemented yet for the Database backends."
+    ## PRIVATE
+
+       Creates a column from a single constant value. The value must be numeric,
+       text, or boolean.
+
+       This method is not static; it must be called on an existing column. That
+       existing column is only used for its internal state; the contents of the
+       column are not used.
+
+       Arguments:
+       - value: the constant value to use for the entire column. Must be
+         numeric, text, or boolean.
+
+       > Example
+         Create a column of the value 42
+
+         column.const 42
+    const : Any -> Column
+    const self value =
+        case value.is_a Number || value.is_a Boolean || value.is_a Text of
+            True ->
+                new_name = "Constant " + value.to_text
+                self.make_binary_op "CONST" value new_name include_self=False
+            False ->
+                msg = "Cannot create a constant column of " + value.to_text
+                Error.throw (Illegal_Argument.Error msg)
+
+    ## Round to a specified number of decimal places.
+
+       By default, rounding uses "symmetric round-half-up", also known as
+       "round towards 0." If use_bankers=True, then it uses "round-half-even",
+       also known as "banker's rounding".
+
+       If `decimal_places` > 0, `round` returns a column of the same type as the
+       input. Otherwise, it returns an `Integer` column.
+
+       Arguments:
+       - decimal_places: The number of decimal places to round to. Can be
+         negative, which results in rounding to positive integer powers of 10.
+         Must be between -15 and 15 (inclusive).
+       - use_bankers: Rounds mid-point to nearest even number.
+
+       ! NaN/Inf
+         `round` behaves differently on different database engines.
+         - Postgres:
+           - NaN: Error
+           - +Inf: Error
+           - -Inf: Error
+         - SQLite:
+           - NaN: Returns Nothing
+           - +Inf: Returns Java `Long.MAX_VALUE`
+           - -Inf: Returns Java `Long.MIN_VALUE`
```

Comment:
These seem wrong to me. I know we cannot be 'fixing' every weird DB feature, and it is 'normal' that there are differences.

But I'm wondering if in this particular case it would be worth to somehow detect this and raise an error like Postgres does. It may not be that simple to do though. @jdunkerley do you think it would be worth it? I really like having guarantees that ensure we avoid data corruption and IMO "rounding +inf" yielding a finite number is a candidate for corrupted results.